### PR TITLE
Make restic backup optional and backwards compatible

### PR DESCRIPTION
**Pull request by tobernguyen**
Dependencies:
- https://github.com/flanksource/template-operator/pull/37

### CODE REVIEW

File: config/templates/postgresql-db.yaml [hunk 1]
```diff
@@ -235,13 +351,13 @@ spec:
               name: postgres-operator-cluster-environment
         s3Bucket:
           - description: '{{.metadata.name}} (backup)'
-            bucket: '{{ kget (print "secret/postgres-operator/backup-postgres-" .metadata.name "-config") "LOGICAL_BACKUP_S3_BUCKET" }}'
+            bucket: '{{.spec.backup.bucket}}'
```
Comment (BrendanGalloway) [inline config/templates/postgresql-db.yaml]:
```suggestion
            bucket: '{{ .spec.backup.bucket | default (kget "secret/postgres-operator/postgres-operator-cluster-environment" "BACKUP_S3_BUCKET") }}'
```
to match the restic definition and allow safe handling of the case where backup.bucket is not specified

File: config/templates/postgresql-db.yaml [hunk 2]
```diff
@@ -74,11 +74,11 @@ spec:
...
-          com.flanksource.infra.logs/processors.1.drop_event.when.contains.message: "no action.  i am the leader with the lock"
-          com.flanksource.infra.logs/processors.2.drop_event.when.contains.message: "INFO: Lock owner:"
-          com.flanksource.infra.logs/processors.3.drop_event.when.contains.message: "INFO: does not have lock"
-          com.flanksource.infra.logs/processors.4.dissect.tokenizer: "%{date} %{time} %{TZ} [%{PID}] %{user} %{level}: %{log}"
+          com.flanksource.infra.logs/processors.0.drop_event.when.contains.message: no action.  i am a secondary and i am following a leader
+          com.flanksource.infra.logs/processors.1.drop_event.when.contains.message: no action.  i am the leader with the lock
+          com.flanksource.infra.logs/processors.2.drop_event.when.contains.message: 'INFO: Lock owner:'
+          com.flanksource.infra.logs/processors.3.drop_event.when.contains.message: 'INFO: does not have lock'
+          com.flanksource.infra.logs/processors.4.dissect.tokenizer: '%{date} %{time} %{TZ} [%{PID}] %{user} %{level}: %{log}'
```
Comment (BrendanGalloway) [inline config/templates/postgresql-db.yaml:81]: what was the motivation for changing these lines?

Comment (tobernguyen) [reply config/templates/postgresql-db.yaml:81]: I just run `make` at the root directory to re-generate the CRDs from go code. But it also ran lint and the linter made these changes.

File: config/templates/postgresql-db.yaml (hunk 1 above)
Comment (tobernguyen) [reply config/templates/postgresql-db.yaml]: This was the original setup prior to my restic implementation so I just revert it to the original one. But yeah I think adding it back should not hurt. Let me add this back.

Comment (tobernguyen) [reply config/templates/postgresql-db.yaml]: Ah, sorry I read too fast. You were suggesting add an improvement, not reverting to the old code. Thanks for the improvement. I added your suggestion!

**Pull request merged**